base/math.jl

```diff
@@ -853,6 +854,21 @@ function exponent(x::T) where T<:IEEEFloat
     return k - exponent_bias(T)
 end
 
+# Like exponent, but assumes the nothrow precondition. For
+# internal use only. Could be written as
+# @assume_effects :nothrow exponent()
+# but currently this form is easier on the compiler.
+function _exponent_finite_nonzero(x::T) where T<:IEEEFloat
```

Comment:
This deserves a better name.

Reply:
It's `exponent`, but only for for finite and nonzero values. Seems perfectly descriptive to me.

Reply:
understood, but given that it's something that I will probably want to use, I think something like `nothrow_exponent` with documentation as to when it gives wrong answers would be better.

Reply:
Since we don't have any precondition enforcement, I do like having the precondition in the name, so it's obvious to people reading it at the callsite what is being asserted here.